### PR TITLE
Allow macro to be nested like function call

### PR DESCRIPTION
**Pull request by topecongiro**
This PR allows macro to be nested like function call:
```rust
foo(try!(
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
))
```
Also, currently rustfmt forgets to add a trailing comma for multiline args when the function call is inside macro:
```rust
try!(foo(
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
));
```
This PR fixed this.

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -74,7 +74,13 @@ fn format_expr(expr: &ast::Expr,
         }
         ast::ExprKind::Call(ref callee, ref args) => {
             let inner_span = mk_sp(callee.span.hi, expr.span.hi);
-            rewrite_call(context, &**callee, args, inner_span, shape)
+            if context.inside_macro {
+                let mut context = context.clone();
+                context.inside_macro = false;
+                rewrite_call(&context, &**callee, args, inner_span, shape)
```
Comment (nrc) [inline src/expr.rs]: I think we should be doing this at a higher level in the stack. What I mean is that we set this true so as not to insert/remove commas, so once we are walking deeper in the tree where changing commas, we should unset it. At the moment, we only check this in function calls, but one could imagine we check it for a different AST node and then we have a bug if we forget to unset the flag there too.

File: src/expr.rs [hunk 2]
```diff
@@ -1776,27 +1776,25 @@ fn rewrite_call_args(context: &RewriteContext,
         acc + item.item.as_ref().map_or(0, |s| 2 + first_line_width(s))
     }) <= one_line_budget + 2;
 
-    match write_list(&item_vec, &fmt) {
+    let result = write_list(&item_vec, &fmt);
+    let last_char = result
+        .as_ref()
+        .map_or(' ', |r| r.chars().last().unwrap_or(' '));
```
Comment (nrc) [inline src/expr.rs]: rather than keeping the last char and (I think) using `' '` as a null value, you could keep a `last_char_is_comma` bool here.

Comment (topecongiro) [inline src/expr.rs]: Nice, thanks!

File: src/expr.rs (hunk 1 above)
Comment (topecongiro) [inline src/expr.rs]: I see, like inside `rewrite_call_args`?

**Comment by topecongiro** [comment]
Updated & rebased suggested changes.

**Pull request merged**